server/games/game.py

```diff
@@ -593,6 +607,9 @@ def clear_slot(self, slot_index):
                 if not ranked:
                     await self.mark_invalid(ValidityState.BAD_MAP)
 
+            else:
+                await self.mark_invalid(ValidityState.BAD_MAP)
```

Comment:
that's a lot of whitespace above

Reply:
you set invalid if result and not ranked but also when no result
--> boolean logic can be shortened if    `not result present OR result not ranked` while evaluating just the not result present if it isn't present (I don't know python logic operators) so you don't try to access a null object.

Reply:
But ranked is not declared on the same level, so it won't be accessable out of the "if result" loop. How am I supposed to solve this ?

Reply:
yes, you somehow need to get the ranked out of result in the same expression, I don't know python, if result is present meaning the first expression evaluated to false meaning the second one will be evaluated, can't you just get that out of the tuple (via index out of an array)?